src/mesh.cpp

```diff
@@ -2004,6 +2039,34 @@ std::string MOABMesh::library() const
   return mesh_lib_type;
 }
 
+// Sample position within a tet for MOAB type tets
+Position MOABMesh::sample(uint64_t* seed, int32_t tet_bin) const {
+
+  moab::EntityHandle tet_ent = get_ent_handle_from_bin(tet_bin);
+
+  // Get vertex coordinates for MOAB tet
+  vector<moab::EntityHandle> conn1;
+  moab::ErrorCode rval = mbi_->get_connectivity(&tet_ent, 1, conn1);
+  if (rval != moab::MB_SUCCESS) {
+    fatal_error("Failed to get tet connectivity");
+  }
+  moab::CartVect p[4];
```

Comment:
Nit pick, but just having 4 is a magic number. It would be better to make a descriptive int variable so that anyone who looks at it later knows why this number. Then you should replace the 4 that appear after with the variable name where it appears. Also, if for some reason this 4 changes in the future, then you only need to replace it in one spot. Otherwise you may miss one and it could cause a hard-to-find error.

Reply:
I agree in principle. I think I will leave this code as is because it matches a similar implementation in a different file and the '4' in this case will never change and should be recognized as the number of vertices on a tet

Reply:
Gotcha, yeah I'd say that is up to you and @pshriwise. It makes sense that four won't change since a tet always has four vertices. If it matches another file, then maybe it does make more sense to leave it, so the code is parallel to itself in style (or change everywhere it appears 😉 )

Reply:
It's true, the number of tet vertices is hard-coded throughout `mesh.cpp`. It would be good to add a constant in `mesh.h` to improve clarity. I can handle that in a follow-on PR if you like @NybergWISC.